doc/develop/test/twister/twister_blackbox.rst

```diff
@@ -0,0 +1,119 @@
+.. _twister_blackbox:
+
+Twister blackbox tests
+######################
+
+This guide aims to explain the structure of a test file so the reader will be able
+to understand existing files and create their own. All developers should fix any tests
+they break and create new ones when introducing new features, so this knowledge is
+important for any Twister developer.
+
+Basics
+******
+
+Twister blackbox tests are written in python, with auxiliary test data following
+whichever format it was in originally. They are wholly contained in the
+`scripts/tests/twister_blackbox/` directory inside zephyr and prepended with `test_`.
+
+Blackbox tests should not be aware of the internal twister code. Instead, they should
+call twister as user would and check the results.
+
+Sample test file
+****************
+
+.. include:: ./sample_blackbox_test.py
+   :literal:
+   :code: python
+   :number-lines:
+
+Comparison with CLI
+*******************
+
+Test above runs the command
+``twister -i --outdir $OUTDIR -T $TEST_DATA/tests -y --level $LEVEL --test-config $TEST_DATA/test_config.yaml -p qemu_x86 -p frdm_k64f`` .
+The flags are set in the ``args`` variable, which is a list of ``str`` 's [#f1]_ .
+The arguments are then provided to Twister in the first ``with`` segment via ``mock`` library.
```

Comment:
please put links to related libraries/methods etc. - like in the second part of this guide.